api/turing/cluster/controller.go

```diff
@@ -64,18 +64,25 @@ type clusterConfig struct {
 // Controller defines the operations supported by the cluster controller
 type Controller interface {
 	DeployKnativeService(ctx context.Context, svc *KnativeService) error
+	IsKnativeServiceInNamespace(svcName string, namespace string) bool
```

Comment:
Instead of `IsXXXXInNamespace`, can we simply add a flag to the existing delete methods called `ignoreNotFound` or something like that and let it selectively return the error or nil when the resource doesn't exist?

We are already doing a get operation inside each of the delete methods, so it may be wasteful to do it once before calling delete as well. Alternatively, we can remove the get operation from within the existing delete methods while keeping these new methods, but I feel that the adding the flag instead will keep the number of methods in the interface concise.

Reply:
The `ignoreNotFound` flag sounds good! I'll refactor the code to use that.